### DISCUSSION TITLE
使用疑问

### DISCUSSION THREAD

**Discussion by HuangAdam** [DreamVoid/MiraiMC#239] [问答 / Q&A]
@DreamVoid 
关于Http API的设置有些疑问，文档写的也不太清楚

1. 看到后台公告上面说暂不使用2.12.0，我MCL用的2.12.0还能用httpapi连接吗？
2. 连接指令”/mirai login <机器人账号> <连接密钥> HTTPAPI“中的连接密钥是什么？
3. MCL是否需要插件mirai-api-http？
4. MCL是否需要有什么对应配置？比如该插件里的url对应MCL在哪里设置

主要问题就这四个

**Comment by DreamVoid** [comment]
不使用2.12.0是miraimc暂时不支持2.12.0的核心，对于mcl是没有要求的。
密钥是你在mcl上的HTTP API插件的密钥，在mcl启动时或者配置文件可以找到。
需要。
mcl不需要什么配置，唯一需要配置的可能是连接密钥。miraimc需要配置目标mcl的URL地址。

**Reply by HuangAdam** [reply]
> 不使用2.12.0是miraimc暂时不支持2.12.0的核心，对于mcl是没有要求的。 密钥是你在mcl上的HTTP API插件的密钥，在mcl启动时或者配置文件可以找到。 需要。 mcl不需要什么配置，唯一需要配置的可能是连接密钥。miraimc需要配置目标mcl的URL地址。

@DreamVoid 

所以连接密钥是Http api配置里的这个吧：
![01BX}0_N5UJ(QI67CC@I~0O](https://user-images.githubusercontent.com/46439030/187120220-e484eb9d-24e7-4141-b3a8-87ff417bc827.png)
需要手动添加个adapters吗？
端口需要转发吗？MCL也在本地
可以关闭连接密钥吗？

**Reply by HuangAdam** [reply]
>  ![01BX}0_N5UJ(QI67CC@I~0O](https://user-images.githubusercontent.com/46439030/187120220-e484eb9d-24e7-4141-b3a8-87ff417bc827.png) 

我的连接密钥就是图片的这个，但是输入指令会报错，就一条：
`[10:47:43 WARN] [MiraiMC]: 登录机器人时出现异常，原因: me.dreamvoid.miraimc.libraries.org.apache.http.client.ClientProtocolException`
不确定是不是我设置错了

**Reply by HuangAdam** [reply]
@DreamVoid 
这个属于bug还是操作错误？

**Reply by DreamVoid** [reply]
> @DreamVoid 
> 这个属于bug还是操作错误？

更新最新开发版
